### PR TITLE
Determinant with small numbers fix

### LINKED ISSUE #3135: determinant of matrix with tiny numbers
`math.det([[6.123234262925839e-17,-1,1],[-0.8660253882408142,0.5,1],[-0.6495190262794495, -0.3749999701976776,1]]);`
returns `0`, while (the only change is in the first number from `e-17` to `e-10`)
`math.det([[6.123234262925839e-10,-1,1],[-0.8660253882408142,0.5,1],[-0.6495190262794495, -0.3749999701976776,1]]);`
returns `0.433` (the correct result).

The first result is mathematically incorrect, and I am not aware of any numerical issues that could be causing this. Therefore, I believe it is a bug.

### PR DESCRIPTION
**Pull request by dvd101x**
This is an attempt to fix #3135, the issue is that isZero yields false when beeing very close to zero.

I found an issue with config.epsilon that will address separately.

### CODE REVIEW

**Comment by josdejong** [comment]
Thanks David. You call this PR an "attempt", but it looks like it fully addresses the rounding issue for `isZero`. Are there other things that you want to address here? I think it probably makes sense to address `isPositive` and `isNegative` in this same PR, so the three functions keep working in a consistent way.

**Comment by dvd101x** [comment]
With this latest commit `isPositive` and `isNegative` now works as discussed including some tests.

File: src/function/utils/isZero.js [hunk 1]
```diff
@@ -40,19 +39,7 @@ export const createIsZero = /* #__PURE__ */ factory(name, dependencies, ({ typed
...
-    },
-
-    Complex: function (x) {
-      return x.re === 0 && x.im === 0
-    },
-
-    Fraction: function (x) {
-      return x.d === 1 && x.n === 0
-    },
+    'number | BigNumber | Complex | Fraction': x => equalScalar(x, 0),
```
Comment (josdejong) [inline src/function/utils/isZero.js:42]: This is a very nice simplification 😄

**Comment by josdejong** [comment]
Thanks, this looks good David!

**Comment by josdejong** [comment]
This originally was to fix #3135. I think it would be good to add a unit test to verify that #3135 is solved now, or would you like to do that in a separate PR?

**Comment by dvd101x** [comment]
I included the test for #3135 
I don't know if it might be too specific.

**Comment by josdejong** [comment]
Looks good to me. It indeed feels like a quite specific test, but I think it's fine (we can always add more tests if needed in the future).

I'll merge this PR now in a new `v13` branch.

**Comment by josdejong** [comment]
Published now in `v13.0.0`, thanks again David!

**Pull request merged**